src/macaron/slsa_analyzer/analyzer.py

```diff
@@ -341,11 +362,66 @@ def run_single(
                 status=SCMStatus.ANALYSIS_FAILED,
             )
 
+        # Prepare the repo.
+        git_obj = None
+        if analysis_target.repo_path:
+            git_obj = self._prepare_repo(
+                os.path.join(self.output_path, self.GIT_REPOS_DIR),
+                analysis_target.repo_path,
+                analysis_target.branch,
+                analysis_target.digest,
+                analysis_target.parsed_purl,
+            )
+
+        # Check if only one of the repo or digest came from direct input, and a versioned purl exists for the other.
+        if (provenance_repo_url or provenance_commit_digest) and parsed_purl and parsed_purl.version:
+            if determine_abstract_purl_type(parsed_purl) == AbstractPurlType.REPOSITORY:
+                if not repo_path_input and provenance_repo_url:
+                    # Check the repo from purl.
+                    possibly_expanded_type = parsed_purl.type
+                    if defaults.has_section("package_url"):
+                        possibly_expanded_type = defaults.get(
+                            "package_url", parsed_purl.type, fallback=parsed_purl.type
+                        )
+                    repo_url_input = urllib.parse.urlparse(repo_path_input)
+                    purl_path = parsed_purl.name
+                    if parsed_purl.namespace:
+                        purl_path = f"{parsed_purl.namespace}/{purl_path}"
+                    if repo_url_input.netloc != possibly_expanded_type or repo_url_input.path != purl_path:
```

Comment:
If I understand correctly, within this if block, we are comparing the repo url obtained from the PURL against the repo url obtained from the provenance. Should this comparison between them instead of the repo url from user input ?